### ISSUE TITLE
Change Newtonsoft.Json to System.Text.Json

### ISSUE THREAD

**Issue by alexsaare** [neutmute/loggly-csharp#112]
Would it be possible to change Newtonsoft.Json to System.Text.Json?  There's performance benefits, it's lighter weight and it keeps the code native to Microsoft.

**Comment by snakefoot** [comment]
Yes but there are different side-effects:
- System.Text.Json doesn't implement reference loop protection.
- System.Text.Json doesn't render double zero as `0.0` but as `0` (Can give funny datatype-guessing issue when indexing)
- System.Text.Json doesn't render fields by default (can be forced on)
- System.Text.Json doesn't recognize the class-attributes from Newtonsoft.Json (for guiding serialization)
- System.Text.Json carries a lot of new dependencies for .NetFramework, and those can be annoying because of strong-version issues.

See also: https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-5-0

**Comment by alexsaare** [comment]
Thanks for the response.  I agree it's a change that requires thought and good communication. 

There are many differences in behaviour however, they are not, IMO, a reason to keep the dependency given that most of them can be controlled with configuration and extension points such as `JsonConverter<T>`s.  

ReferenceLoopHandling would be my only concern for this library and even then, to differ from the System.Text.Json implementation it's a setting which has to be explicitly set to Ignore (which this library does not do by default and I don't believe can be set via LogglyClient - please correct me if i'm wrong).

**Comment by snakefoot** [comment]
The topic says change (replace) one with the other. I'm saying that you probably want to keep NewtonSoft around. And System.Text.Json is not a good dependency for net45. But if new net50 target is added, then one could have both with a setting (or maybe a rewrite is better)


Sent from my Sony Xperia


---- alexsaare wrote ----


Thanks for the response. I agree it's a change that requires thought and good communication.

There are many differences in behaviour however, they are not, IMO, a reason to keep the dependency given that most of them can be controlled with configuration and extension points such as JsonConverter<T>s.

ReferenceLoopHandling would be my only concern for this library and even then, to differ from the System.Text.Json implementation it's a setting which has to be explicitly set to Ignore (which this library does not do by default and I don't believe can be set via LogglyClient - please correct me if i'm wrong).

**Comment by alexsaare** [comment]
Agreed - for net45 (and netstandard 1.6) I would keep Newtonsoft and have a multi target build which uses System.Text.Json for anything netstandard2.0+.
